spec/requests/spree/admin/overview_spec.rb

```diff
@@ -13,7 +13,11 @@
 
   describe "GET /admin" do
     before do
-      allow(TermsOfServiceFile).to receive(:updated_at).and_return(2.hours.ago)
+      mocked_tos = double(TermsOfServiceFile, updated_at: 2.hours.ago)
+      allow(TermsOfServiceFile).to receive(:current).and_return(mocked_tos)
+      # Mock current_url so we don't have to set up a complicated TermsOfServiceFile mock
+      # with attachement
+      allow(TermsOfServiceFile).to receive(:current_url).and_return("tmp/tos.pdf")
```

Comment:
Sorry I'm just wondering why this is needed in the overview spec, when it looks like we also test the same conditions in `spec/system/admin/tos_banner_spec.rb`?

Reply:
It's because of the implementation of `TermsOfServiceFile.current_url` : 
```ruby
def self.current_url
    Rails.application.routes.url_helpers.url_for(current.attachment) if current
end
```
Now `TermsOfServiceFile.current` returns something so the code tries create a url, we just mock it here as we are not really interested in testing the url is correct.